mmv1/third_party/terraform/transport/transport.go

```diff
@@ -189,3 +189,140 @@ func IsApiNotEnabledError(err error) bool {
 	}
 	return false
 }
+
+type GetPaginatedItemsSliceOptions struct {
+	ResourceData   *schema.ResourceData
+	Config         *Config
+	BillingProject *string
+	UserAgent      string
+	URL            string
+	ListFlattener  func(config *Config, res interface{}) ([]interface{}, error)
+	Params         map[string]string
+	ResourceToList string
+}
+
+func GetPaginatedItemsSlice(paginationOptions GetPaginatedItemsSliceOptions) ([]interface{}, error) {
```

Comment:
As far as I can tell this just returns a weaker type than GetPaginatedItemsMapOptions in all cases. Could we just strengthen the type in the callers? (We also appear to be weakening the type in AlloyDB locations right now)